pkg/apis/serving/v1beta1/inference_service_validation.go

```diff
@@ -252,9 +261,19 @@ func validateInferenceServiceAutoscaler(isvc *InferenceService) error {
 	return nil
 }
 
+// Validate of autoscaler KEDA metrics
+func validateKEDAMetrics(metric ScaleMetric) error {
+	for _, item := range constants.AutoscalerAllowedKEDAMetricsList {
+		if item == constants.AutoscalerMetricsType(metric) {
+			return nil
+		}
+	}
+	return fmt.Errorf("[%s] is not a supported metric in KEDA.\n", metric)
+}
+
 // Validate of autoscaler HPA metrics
 func validateHPAMetrics(metric ScaleMetric) error {
-	for _, item := range constants.AutoscalerAllowedMetricsList {
+	for _, item := range constants.AutoscalerAllowedHPAMetricsList {
```

Comment:
In Go 1.18 and later, you can use the slices package from the standard library to check if an item exists in a list (slice).
you can use the package slices as well here.